File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        output = defaultdict(list)
        for word in strs:
            count = [0] * 26
            for c in word:
                count[ord(c) - ord('a')] += 1
            output[tuple(count)].append(word)
        return list(output.values())

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> hashmap = new HashMap<>();
        for (String word: strs) {
            int[] count = new int[26];
            for (char c : word.toCharArray()) {
                count[c-'a'] += 1;
            }
            String key = Arrays.toString(count);
            hashmap.putIfAbsent(key, new ArrayList<String>());
            hashmap.get(key).add(word);
        }
        return new ArrayList<>(hashmap.values());
    }
}